### PR TITLE
Added auto detection for kubelet client config

### PR DESCRIPTION
**Pull request by cristianciutea**
This PR will add the capability to auto-detect the kubelet endpoint configuration by fetching the node information.
If this mechanism will be used only if no configuration is specified via cmd line args.
If this mechanism fails, the old behavior is use.
In addition to this, a new flag is added to be able to overwrite the hostname
--host (default localhost). If not specified the previous behavior is maintained.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -10,22 +10,33 @@ import (
...
 	envPrefix  = "NRIA"
+
+	DefaultHost = "localhost"
 )
 
 var _ = flag.String(namespaces, "", "(optional, default '') Comma separated list of namespaces to discover pods on")
 var _ = flag.Bool(insecure, false, `(optional, default false, deprecated) Use insecure (non-ssl) connection.
 For backwards compatibility this flag takes precedence over 'tls')`)
+var _ = flag.Bool(autoConfig, false, "(optional, default false) fetch node info for configuration")
+var _ = flag.Int(timeout, 5000, "(optional, default 5000) timeout in ms")
```
Comment (brushknight) [inline internal/config/config.go:27]: do we need to mention this new flag in PR description + release notes?

File: internal/http/client.go [hunk 2]
```diff
@@ -47,3 +53,39 @@ func NewClient(url url.URL, token string) HttpClient {
...
+	resp, err := kc.client.Do(netHttp.MethodGet, path)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != netHttp.StatusOK {
+		return nil, errors.New(resp.Status)
+	}
+
+	buff, _ := ioutil.ReadAll(resp.Body)
+	_ = resp.Body.Close()
```
Comment (brushknight) [inline internal/http/client.go:89]: why do we need this blank variable?

Comment (carlosroman) [reply internal/http/client.go:89]: It makes linters happy when you assign errors to a blank variable rather than just ignoring them.

Comment (carlosroman) [reply internal/http/client.go:89]: I'm more concerned we don't check the error from `ioutil.ReadAll` or at least make it part of the return on line 90.

File: internal/http/client.go [hunk 3]
```diff
@@ -47,3 +53,39 @@ func NewClient(url url.URL, token string) HttpClient {
...
+	}, nil
+}
+
+func (kc *kubeletClient) Get(path string) ([]byte, error) {
+	resp, err := kc.client.Do(netHttp.MethodGet, path)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != netHttp.StatusOK {
+		return nil, errors.New(resp.Status)
```
Comment (carlosroman) [inline internal/http/client.go:85]:
```suggestion
                _, _ = ioutil.ReadAll(r.Body) // read all body to reuse resources
		return nil, errors.New(resp.Status)
```

File: internal/http/client.go [hunk 4]
```diff
@@ -47,3 +53,39 @@ func NewClient(url url.URL, token string) HttpClient {
...
+		client: client,
+	}, nil
+}
+
+func (kc *kubeletClient) Get(path string) ([]byte, error) {
+	resp, err := kc.client.Do(netHttp.MethodGet, path)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != netHttp.StatusOK {
```
Comment (carlosroman) [inline internal/http/client.go:84]:
```suggestion
        defer resp.Body.Close()
	if resp.StatusCode != netHttp.StatusOK {
```

**Review by haooliveira84 — APPROVED** [review]
lgtm

File: internal/config/config.go [hunk 5]
```diff
@@ -35,14 +44,32 @@ func splitStrings(str string) []string {
...
+	found := false
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == name {
+			found = true
+		}
+	})
+	return found
+}
+
+// IsAutoConfig returns true if no config parameter was provided as cmd line arg.
```
Comment (carlosroman) [inline internal/config/config.go:58]: Cool 👍

**Review by carlosroman — APPROVED** [review]
LGTM, thanks for making the HTTP changes as well.

**Pull request merged**